IM_calculation/Advanced_IM/Models/Burks_Baker_2013/run.py

```diff
@@ -0,0 +1,140 @@
+"""
+Python script to run a 3D waveform through Burks_Baker_2013_elastic_inelastic and store the outputs to a txt file
+"""
+
+import argparse
+
+import logging
+from pathlib import Path
+
+import numpy as np
+import pandas as pd
+
+from IM_calculation.Advanced_IM import runlibs_2d
+from IM_calculation.IM.intensity_measures import get_geom
+from IM_calculation.IM import intensity_measures
+from IM_calculation.IM import im_calculation
+
+from qcore.timeseries import read_ascii
+from qcore import constants
+from qcore.constants import Components
+
+component_list = [Components.c000, Components.c090]
+rotd_comps = [Components.crotd50, Components.crotd100, Components.crotd100_50]
+STORIES = 10
+
+z_list = [0.05]  # damping ratio
+alpha_list = [0.05]  # strain hardening ratio
+dy_list = [0.1765, 0.4, 0.6]  # strain hardening ratio
+dt_list = [0.005]
+period = np.array(constants.DEFAULT_PSA_PERIODS)
+
+
+def main(comp_000: Path, comp_090: Path, rotd: bool, output_dir: Path):
+    """
+
+    Parameters
+    ----------
+    comp_000 : Path to 000 component waveform file
+    comp_090 : Path to 090 component waveform file
+    rotd : True if rotd components need to be computed
+    output_dir : Path to the output directory
+    """
+
+    output_dir.mkdir(exist_ok=True)
+
+    log_file = output_dir / "log"
+    logging.basicConfig(
+        format="%(asctime)s %(message)s", filename=log_file, level=logging.DEBUG
+    )
+
+    waveforms = {}
+    waveforms["000"], meta = read_ascii(comp_000, meta=True)
+    waveforms["090"] = read_ascii(comp_090)
+    DT = np.float32(meta["dt"])  # to match the behaviour of basic SDI
+    NT = meta["nt"]
+
+    results = {}
+
+    accelerations = np.array((waveforms["000"], waveforms["090"])).T
+    ordered_columns_dict = {}
+
+    for z in z_list:
+        for dt in dt_list:
+            for alpha in alpha_list:
+                for dy in dy_list:
+                    displacements = (
+                        intensity_measures.get_SDI_nd(
+                            accelerations, period, NT, DT, z, alpha, dy, dt
+                        )
+                        * 100  # Burks & Baker returns m, but output is stored in cm
+                    )
+                    sdi_values = np.max(np.abs(displacements), axis=1)
+                    im_names = []
+                    for t in period:
+                        im_name = f"SDI_{t}_dy{dy}_a{alpha}_dt{dt}_z{z}"
+                        im_names.append(im_name)
+                        # for each period, we will have a list of im_names varying dy,alpha,dt and z, to help ordering columns later
+                        if t not in ordered_columns_dict:
+                            ordered_columns_dict[t] = []
+                        ordered_columns_dict[t].append(im_name)
```

Comment:
my favourite way of doing this block is ```ordered_columns_dict.get(t, []).append(im_name).```

But happy as is; it seems that there are a few instances where you do the check, create, update paradigm.